engine/schema/src/main/java/com/cloud/capacity/dao/CapacityDaoImpl.java

```diff
@@ -903,20 +903,28 @@ public boolean removeBy(Short capacityType, Long zoneId, Long podId, Long cluste
     }
 
     @Override
-    public List<Long> orderHostsByFreeCapacity(Long clusterId, short capacityTypeForOrdering){
+    public List<Long> orderHostsByFreeCapacity(Long zoneId, Long clusterId, short capacityTypeForOrdering){
```

Comment:
@weizhouapache wouldn't a cluster imply a zone? we wouldn't have to pass a zone if we already know the cluster.

Reply:
@DaanHoogland if we search storage pools by clusterid, we would not get the list of zone-wide storage pools, because cluster_id is NULL.

Reply:
makes sense, but if we have a cluster we can find the zone it is in and use that to find zone wide pools for it?

Reply:
to be clear, I think this is aesthetic, not a potential bug.

Reply:
@DaanHoogland 
please look at the code, there is only 'AND cluster_id = ' in the sql.
you can also execute the query in mysql
```
select * from op_host_capacity where capacity_type=3 and cluster_id=?? 
```
it lists only the cluster-wide storage pools.

Reply:
yes, i read the code. Either zone_id or cluster_id should be null. What is the use case of both being not null?
My point is; pass the cluster and find all storage pools for that cluster either cluster wide or for the zone the cluster is in.

Reply:
@DaanHoogland what if the storage pool is used for another zone ?

Reply:
if we pass the cluster and deduce the zone from that, or if we deliberately pass another zone than the cluster we are looking to deploy in? I don't see how can it ever be another zone,except for a bad bug. Or do you mean a storage pool that is used for multiple zones? I have not seen that before.

Reply:
@DaanHoogland as I said, if we pass the cluster id, it lists only cluster-wide storage pools.
To list zone-wide storage pools, we should pass the zoneId/data_center_id (and cluster_id should be NULL).